### PR TITLE
added incoming friends request retrieval and current friend

### PR DESCRIPTION
**Pull request by clintBock**

### CODE REVIEW

File: resources/views/groups.blade.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
Comment (nikoootine) [inline resources/views/groups.blade.php]: usually it's the FE people that creates the blade files. This shouldn't be included in the push. same as the friends view

File: app/User.php [hunk 2]
```diff
@@ -61,9 +61,32 @@ public function allFriends()
...
+     */
+    public function friendsRequests()
+    {
+        $foundInUser1 =  $this->belongsToMany('App\User', 'friends', 'username1', 'username2')
+                              ->wherePivot('status_code', '=', 'pending')
+                              ->wherePivot('action_username', '!=', $this->username)->get();
+        $foundInUser2 =  $this->belongsToMany('App\User', 'friends', 'username2', 'username1')
+                              ->wherePivot('status_code', '=', 'pending')
+                              ->wherePivot('action_username', '!=', $this->username)->get();
+        $merged = $foundInUser1->merge($foundInUser2);
```
Comment (nikoootine) [inline app/User.php]: The query is wrong, this shouldn't be both ways. You only need to retrieve the requests that are sent to the user.

File: app/User.php [hunk 3]
```diff
@@ -61,9 +61,32 @@ public function allFriends()
...
     {
-        return $this->belongsToMany('App\User', 'friends', 'username1', 'username2')
-                    ->wherePivot('status_code', '=', 'accepted');
+        $foundInUser2 = $this->belongsToMany('App\User', 'friends', 'username2', 'username1')
+            ->wherePivot('status_code', '=', 'accepted')->get();
+        $foundInUser1 = $this->belongsToMany('App\User', 'friends', 'username1', 'username2')
+                             ->wherePivot('status_code', '=', 'accepted')->get();
+
+        $merged = $foundInUser1->merge($foundInUser2);
+        return $merged;
```
Comment (nikoootine) [inline app/User.php]: Smooooooooooooooooooooth

File: app/Http/Controllers/GroupController.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace App\Http\Controllers;
+use DB;
+use Auth;
+use User;
+use Illuminate\Http\Request;
+
+class GroupController extends Controller
+{
+    public function Index(){
```
Comment (nikoootine) [inline app/Http/Controllers/GroupController.php]: can we make the first letters small lol

File: app/GroupExpense.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class GroupExpense extends Model
+{
+    public function Group(){
+        return $this->belongsto('App\Group', 'group_id');
```
Comment (nikoootine) [inline app/GroupExpense.php]: Does this work? The method should belongsTo should have a capital 'T'

File: app/Http/Controllers/FriendController.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+namespace App\Http\Controllers;
+use DB;
+use Auth;
+use User;
+use Illuminate\Http\Request;
+
+class FriendController extends Controller
+{
+    public function Index(){
+        $user = \App\User::find(Auth::user() -> username);
```
Comment (nikoootine) [inline app/Http/Controllers/FriendController.php]: I think you can just do $user = Auth::user();. But im not sure

File: app/GroupExpense.php (hunk 5 above)
Comment (clintBock) [inline app/GroupExpense.php]: uh maybe not, but this all needs to be redone anyway its not finished right

File: app/Http/Controllers/GroupController.php (hunk 4 above)
Comment (clintBock) [inline app/Http/Controllers/GroupController.php]: lol

File: app/User.php (hunk 3 above)
Comment (clintBock) [inline app/User.php]: whats this

File: app/Http/Controllers/FriendController.php (hunk 6 above)
Comment (itsmattli) [inline app/Http/Controllers/FriendController.php]: if you're looking for the current user, just Auth::user()->username will do.

if you want to search a list of users. Add `use App\User;` at the top then you can just do User::find

**Comment by nikoootine** [comment]
I would prefer these relational queries in the model. They best describe what the relationship of one model has with other models. "[Model] belongs to many ...", "[Model] has many .."

**Pull request merged**